### PR TITLE
Inserted Decorators

### PR DESCRIPTION
**Pull request by NatanimA**
#### Interface
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn your Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### See your decorators in action
\Try the following code and check if you managed to decorate your person:

``` ruby
person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name
```
### The output is as follows

ruby

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -18,4 +21,15 @@ def of_age?
   def can_use_services?
     of_age? || @parent_permission ? true : false
   end
+
+  def correct_name
+    @name
+  end
 end
+# #Check Using this code
+# person = Person.new(22, 'maximilianus')
```
Comment (HAMMAS-SALEEM) [inline person.rb:30]: image


- [ ] You have a great readme file 👏 but this link is not working 😨 Kindly fix this link to make your readme file descriptive 😊 and to make your license file easily accessible for all the users 👍

Comment (HAMMAS-SALEEM) [inline person.rb:30]: image


- [ ] It's great that you have a license file in your project but it looks like a template 🤔 Kindly add your **name**, **current year** and **app-type** 😊 to make your project licensed to you 👍

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @NatanimA :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -282,7 +282,7 @@ I would like to thank...
 
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
+This project is [MIT](./MIT) licensed.
```
Comment (DammyShittu) [inline README.md]: - [ ] When I clicked the MIT link see the error I got. I think you can fix this by appending `.md` to the link.

  ```
  ...(./MIT.md)...
  ```

  cjcj

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @NatanimA,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @NatanimA 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good design 💯 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**